### PR TITLE
Added HTML parsing for content from Threatmatch

### PR DESCRIPTION
**Pull request by pietrocapece**
### Proposed changes

* Added a function that parses the content being fetched from Threatmatch and removes any HTML tags

### Related issues

* None

### Checklist

- [X] I consider the submitted work as finished
- [X] I tested the code for its functionality using different use cases
- [X] I added/update the relevant documentation (either on github or on notion)
- [X] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: external-import/threatmatch/src/threatmatch.py [hunk 1]
```diff
@@ -81,6 +82,22 @@ def get_interval(self):
...
+            def handle_data(self, data):
+                self.fed.append(data)
+
+            def get_data(self):
+                return "".join(self.fed)
+
+        parser = HTMLTagRemover()
+        parser.feed(text)
+        return parser.get_data()
+
```
Comment (flavienSindou) [inline external-import/threatmatch/src/threatmatch.py]: I think this is not necessary as you could use popular BeatifulSoup library later in the code to remove all tags
```suggestion
                    
``` 
``` 
object["description"] =  bs4.BeautifulSoup(object["description"], "html.parser")..get_text()
```

File: external-import/threatmatch/src/threatmatch.py [hunk 2]
```diff
@@ -101,12 +118,18 @@ def _get_item(self, token, type, item_id):
             headers=headers,
         )
         if r.status_code != 200:
-            self.helper.log_error(str(r.text))
+            self.helper.log_error(
+                f"Could not fetch item: {item_id}, Error: {str(r.text)}"
```
Comment (flavienSindou) [inline external-import/threatmatch/src/threatmatch.py]:
```suggestion
                f"Could not fetch item: {str(item_id)}, Error: {r.text}"
```

**Review by flavienSindou — APPROVED** [review]
I have reviewed and tested part of this implementation, and it works well overall.

I left a few suggestions that I believe would help make the code simpler.

Thank you for adding information and error logs.

The connector still follows an old template of the connector implementation. New features, such as `pycti.OpenCTIConnectorHelper.helper.schedule_iso`, could help you manage the connector's scheduled runs, for instance.

**Comment by helene-nguyen** [comment]
@pietrocapece Thank you for your contribution. Could you resolve the conflicts ?

**Comment by helene-nguyen** [comment]
@pietrocapece Merging still block. Could you sign your commits please?

**Pull request closed without merging**